sdk/scala-sdk/src/main/scala/kalix/scalasdk/impl/DeferredCallAdapters.scala

```diff
@@ -48,6 +50,27 @@ private[scalasdk] final case class ScalaDeferredCallAdapter[I, O](javaSdkDeferre
     new MetadataImpl(javaSdkDeferredCall.metadata.asInstanceOf[kalix.javasdk.impl.MetadataImpl])
 
   def execute(): Future[O] = javaSdkDeferredCall.execute().asScala
+
+  override def withMetadata(entries: Seq[MetadataEntry]): ScalaDeferredCallAdapter[I, O] = {
+    val updatedDefCall = javaSdkDeferredCall.withMetadata(
+      entries
+        .map(entry =>
+          new kalix.javasdk.Metadata.MetadataEntry {
```

Comment:
Maybe we should replace `MetadataEntry` interface with a class because, from a user perspective, it's quite tedious to always provide the same MetadataEntry implementation.

Reply:
See previous comment, the JVM SDK Metadata APIs already provides a smoother API with `add(String, String)` etc.